core/src/main/scala/kafka/coordinator/group/GroupCoordinatorAdapter.scala

```diff
@@ -388,18 +393,27 @@ private[group] class GroupCoordinatorAdapter(
       case retentionTimeMs => Some(currentTimeMs + retentionTimeMs)
     }
 
-    val partitions = new mutable.HashMap[TopicPartition, OffsetAndMetadata]()
+    val topicIds = metadata.topicNamesToIds()
+    val partitions = new mutable.HashMap[TopicIdPartition, OffsetAndMetadata]()
     request.topics.forEach { topic =>
-      topic.partitions.forEach { partition =>
-        val tp = new TopicPartition(topic.name, partition.partitionIndex)
-        partitions += tp -> createOffsetAndMetadata(
-          currentTimeMs,
-          partition.committedOffset,
-          partition.committedLeaderEpoch,
-          partition.committedMetadata,
-          partition.commitTimestamp,
-          expireTimeMs
-        )
+      val topicId = topicIds.get(topic.name)
+      if (Uuid.ZERO_UUID.equals(topicId)) {
+        val topicPartitions = topic.partitions.asScala.map { p =>
+          new TopicPartition(topic.name, p.partitionIndex) -> Errors.UNKNOWN_TOPIC_OR_PARTITION
+        }
+        appendToResponse(topicPartitions.toMap)
+      } else {
```

Comment:
This is not needed as well as it will be done in the KafkaApis layer. We should just take the topic id that we receive here and use it to create the `TopicIdPartition`. Obviously, it will always be `ZERO_UUID` at the moment.

Reply:
I see, I thought we would have to use the resolved id instead of the zero topic id. Hence this change the approach entirely. Thanks for the quick check, I will fix this.